### PR TITLE
add decode_epochs_tmin and decode_epochs_tmax params

### PR DESCRIPTION
**Pull request by agramfort**
### Before merging …

- [ ] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -38,6 +38,7 @@ dependencies = [
     "autoreject",
     "mne[hdf5]",
     "mne-bids[full]",
+    "filelock",
```
Comment (agramfort) [inline pyproject.toml:41]: this is required not just for testing

**Comment by agramfort** [comment]
@hoechenberger should add something like this for full epochs CSP decoding? for now it does 

https://github.com/mne-tools/mne-bids-pipeline/blob/main/mne_bids_pipeline/steps/sensor/_05_decoding_csp.py#L333

**Comment by hoechenberger** [comment]
Thanks @agramfort!

- Should this apply to CSP too?
- I think we should mention the time spans used for decoding in the report somewhere, possibly the figure captions?

**Comment by hoechenberger** [comment]
@agramfort Only now saw your comment above. Do you suggest we have a similar (but separate) setting for CSP?

**Comment by hoechenberger** [comment]
Ah no, for CSP we already **have** a setting:

https://github.com/mne-tools/mne-bids-pipeline/blob/59418291ebafb7a7db5d8615db6e73f2ca8397e5/mne_bids_pipeline/steps/sensor/_05_decoding_csp.py#L282

**Comment by agramfort** [comment]
@hoechenberger let s merge already this and let's see for CSP later

**Comment by hoechenberger** [comment]
One last thing, shouldn't we call the settings `decoding_*`? I think all the others have this prefix – except for the setting called `decode`, which is a bit unfortnutate

File: docs/source/settings/sensor/statistics.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 ::: mne_bids_pipeline._config.contrasts
 ::: mne_bids_pipeline._config.report_evoked_n_time_points
 ::: mne_bids_pipeline._config.decode
+::: mne_bids_pipeline._config.decode_epochs_tmin
+::: mne_bids_pipeline._config.decode_epochs_tmax
```
Comment (agramfort) [inline docs/source/settings/sensor/statistics.md:5]:
```suggestion
::: mne_bids_pipeline._config.decoding_epochs_tmin
::: mne_bids_pipeline._config.decoding_epochs_tmax
```

File: mne_bids_pipeline/_config.py [hunk 3]
```diff
@@ -1325,6 +1325,18 @@
 conditions.
 """
 
+decode_epochs_tmin: Optional[float] = 0.
```
Comment (agramfort) [inline mne_bids_pipeline/_config.py:1328]:
```suggestion
decoding_epochs_tmin: Optional[float] = 0.
```

File: mne_bids_pipeline/_config.py [hunk 4]
```diff
@@ -1325,6 +1325,18 @@
...
+"""
+The first time sample to use for full epochs decoding. By default it starts
+at 0. If None, it starts at the beginning of the epoch.
+"""
+
+decode_epochs_tmax: Optional[float] = None
```
Comment (agramfort) [inline mne_bids_pipeline/_config.py:1334]:
```suggestion
decoding_epochs_tmax: Optional[float] = None
```

File: mne_bids_pipeline/steps/sensor/_02_decoding_full_epochs.py [hunk 5]
```diff
@@ -83,6 +83,7 @@ def run_epochs_decoding(
 
     epochs = mne.read_epochs(in_files.pop('epochs'))
     _restrict_analyze_channels(epochs, cfg)
+    epochs.crop(cfg.decode_epochs_tmin, cfg.decode_epochs_tmax)
```
Comment (agramfort) [inline mne_bids_pipeline/steps/sensor/_02_decoding_full_epochs.py:86]:
```suggestion
    epochs.crop(cfg.decoding_epochs_tmin, cfg.decoding_epochs_tmax)
```

**Comment by hoechenberger** [comment]
Thanks @agramfort

**Pull request merged**